src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,104 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final short RESIZE_MULTIPlIER = 2;
+
+    private Node<K, V>[] table = new Node[INITIAL_CAPACITY];
+    private int threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    private int size = 0;
 
     @Override
     public void put(K key, V value) {
+        Node<K, V> node = getNode(key);
+        if (node != null) {
+            node.value = value;
+            return;
+        }
+
+        if (++size > threshold) {
+            resize();
+        }
 
+        int index = key == null ? 0 : Math.abs(key.hashCode()) % table.length;
+        putNodeByIndex(table, index, new Node<>(key, value));
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> node = getNode(key);
+        return node == null ? null : node.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
+    private Node<K,V> getNode(K key) {
+        int arrayIndex = 0;
+        if (key != null) {
+            arrayIndex = Math.abs(key.hashCode()) % table.length;
+        }
+
+        Node<K, V> node = table[arrayIndex];
+
+        while (node != null) {
+            if ((node.key == null && key == null)
+                    || (
+                    (node.key != null && key != null)
+                    && (node.key.hashCode() == key.hashCode())
+                    && node.key.equals(key)
+                )
+            ) {
+                return node;
+            }
+            node = node.next;
+        }
+
+        return null;
+    }
+
+    private void resize() {
+        final int newSize = table.length * RESIZE_MULTIPlIER;
+        final Node<K, V>[] newTable = new Node[newSize];
+        threshold = (int) (newTable.length * LOAD_FACTOR);
+
+        for (Node<K, V> cell: table) {
+            Node<K, V> node = cell;
+            while (node != null) {
+                Node<K, V> nextNode = node.next;
+
+                int arrayIndex = Math.abs(node.key.hashCode()) % newSize;
+                node.next = null;
+                putNodeByIndex(newTable, arrayIndex, node);
+                node = nextNode;
+            }
+        }
+        table = newTable;
+    }
+
+    private void putNodeByIndex(Node<K, V>[] targetTable, int index, Node<K, V> node) {
```

Comment:
You already have access to your table because it is a class field. Better use it.
```suggestion
    private void putNodeByIndex(int index, Node<K, V> node) {
```

Reply:
I use this function for different tables. In the first case - put the node in the class field table. In the second case - put the node in a new table in resize method. For this purpose, I need the targetTable as an argument of the function.